src/machine-learning/machine-learning-api-client.ts

```diff
@@ -200,15 +213,95 @@ export class MachineLearningApiClient {
       });
   }
 
+  /**
+   * Handles a Long Running Operation coming back from the server.
+   *
+   * @param op The operation to handle
+   * @param options The options for polling
+   */
+  public handleOperation(
+    op: OperationResponse,
+    options?: {
+      wait?: boolean;
+      maxTimeMillis?: number;
+      baseWaitMillis?: number;
+      maxWaitMillis?: number;
+    }):
+    Promise<ModelResponse> {
+    if (op.done) {
+      if (op.response) {
+        return Promise.resolve(op.response);
+      } else if (op.error) {
+        const err = FirebaseMachineLearningError.fromOperationError(
+          op.error.code, op.error.message);
+        return Promise.reject(err);
+      }
+
+      // Done operations must have either a response or an error.
+      throw new FirebaseMachineLearningError('invalid-server-response',
+        'Invalid operation response.');
+    }
+
+    // Operation is not done
+    const opName = op.name!;
+    const metadata = op.metadata || {};
+    const metadataType: string = metadata['@type'] || '';
+    if (!metadataType.includes('ModelOperationMetadata')) {
+      throw new FirebaseMachineLearningError('invalid-server-response',
+        `Unknown Metadata type: ${JSON.stringify(metadata)}`);
+    }
+
+    if (!options || !options.wait) {
+      const modelId = extractModelId(metadata.name);
+      return this.getModel(modelId);
+    }
+
+    return this.pollOperationWithExponentialBackoff(opName, options);
```

Comment:
opName is first used here. May be just remove the variable and use `op.name!` directly?

Reply:
done